### ISSUE TITLE
Is the generator loss calculation wrong?

### ISSUE THREAD

**Issue by vfan26** [nipponjo/deepfillv2-pytorch#30]
In the original tensorflow implementation, the output of inpainted is used to calculate g_loss, but this repository uses the output of stage2 to calculate g_loss.

```
batch_gen = batch_predicted
batch_gen = torch.cat((batch_gen, torch.tile(
mask, [config.batch_size, 1, 1, 1])), dim=1)
d_gen = discriminator(batch_gen)
g_loss = gan_loss_g(d_gen)
```

In the original tensorflow implementation, see [https://github.com/JiahuiYu/generative_inpainting/blob/master/inpaint_model.py#L163](url)

**Comment by nipponjo** [comment]
Hello, good point, thanks for pointing this out. I have corrected it now.
